enhancements/hypershift/kubevirt-static-ip-address-management.md

```diff
@@ -0,0 +1,531 @@
+
+**Control plane nodes or capk controller pods restart**:
+When the capk controller starts, it will check all the virtual machines instances that belongs to it and will reconstruct
+the in memory ip pool, the reconciliation process cannot start until this is finished.
+
+**OpenStack Network Config Format**:
+As stated by the [cloud-init reference docs](https://cloudinit.readthedocs.io/en/latest/reference/network-config.html#network-configuration-sources) the implementation should use the OpenStack network config standard format when cloud init is delivered by config drive. The network configuration is generated as JSON and injected into the VM's cloud-init config drive.
+
+Example OpenStack network config format following the [schema](https://docs.openstack.org/nova/latest/_downloads/9119ca7ac90aa2990e762c08baea3a36/network_data.json):
+```json
+{
+  "links": [
+    {
+      "id": "eno101",
+      "type": "phy",
+      "ethernet_mac_address": "52:54:00:12:34:56"
+    }
+  ],
+  "networks": [
+    {
+      "id": "network0",
+      "type": "ipv4",
+      "link": "eno101",
+      "ip_address": "192.168.1.100", <-- this is the part allocated by "cluster api provider kubevirt controller"
+      "netmask": "255.255.255.0",
+      "routes": [
+        {
+          "network": "0.0.0.0",
+          "netmask": "0.0.0.0",
+          "gateway": "192.168.1.1"
+        }
+      ],
+      "network_id": "network0"
+    }
+  ],
+  "services": [
+    {
+      "type": "dns",
+      "address": "8.8.8.8"
+    },
+    {
+      "type": "dns",
+      "address": "8.8.4.4"
+    }
+  ]
+}
+```
+
+**Afterburn Integration**:
+Production-ready PR implementing this:
+- https://github.com/coreos/afterburn/pull/1238
+
+Afterburn is modified to:
+- Detect when running on KubeVirt (similar to existing ProxmoxVE provider detection)
+- Read OpenStack network config from the cloud-init config drive metadata location (a drive with label `config-2`)
+- Parse OpenStack network config standard JSON format (the file `network_data.json` at the `config-2` drive)
+- Support static IPv4/IPv6 addresses, default gateways, physical ethernet interfaces, DHCP configuration, and DNS nameservers
+- Generate dracut kernel arguments for static network configuration
+- Expose first interface's IPv4 and IPv6 addresses as metadata attributes
+- Preload the virtio_blk module via a dracut module for early boot disk access needed for rhcos + kubevirt (fcos is fine)
+
+**Limitations**:
+- No automatic IP conflict detection across multiple management clusters
+- No integration with external IPAM systems
+
+**Improvements**:
+- If we manage to merge the [PR](https://github.com/coreos/ignition/pull/2134) to use NoCloud with ignition instead of ConfigDrive
+  then the network data format can be cloud init network config v1 or v2 and also JSON or yaml, this is more human friendly than
+  current opnstack config format.
```

Comment:
this was merged already. 

Shouldn't the proposed implementation be revised, and another alternative proposed ?

Reply:
IMHO that information should be called out in the enhancement